sharding/node/backend.go

```diff
@@ -252,21 +244,13 @@ func (s *ShardEthereum) registerActorService(config *params.Config, actor string
 			var txPool *txpool.TXPool
 			ctx.RetrieveService(&txPool)
 			return proposer.NewProposer(config, smcClient, p2p, txPool, shardChainDB.DB(), shardID)
+		} else if actor == "simulator" {
```

Comment:
This looks like a great place for a switch statement. Please change to switch statement where observer is the default.